server/src/routes/api/v1/ingredientsRouter.js

```diff
@@ -53,4 +53,88 @@ ingredientsRouter.post("/", async (req, res) => {
+              description: cleanedIngredient.description
+            })
+          } else {
+            newMeasurement = await PantryMeasurement.query(trx).insertAndFetch({
+              userId: userId,
+              ingredientId: existingIngredient.id,
+              amount: cleanedIngredient.amount,
+              unit: cleanedIngredient.unit,
+              description: cleanedIngredient.description
+            })
+          }
+          return newMeasurement
+        })
+
+        const updatedIngredients = await User.relatedQuery("ingredients").for(userId).where({ name: cleanedIngredient.name })
+        const serializedIngredient = IngredientSerializer.getIngredientWithDetails(updatedIngredients[0])
+        return res.status(201).json({ editedIngredient: serializedIngredient })
+      }
+
+      const relatedIngredient = await User.transaction(async (trx) => {
+        const measurement = await User.relatedQuery("pantryMeasurements", trx).for(userId).where("ingredientId", ingredientId)
+        const deleted = await PantryMeasurement.query(trx).deleteById(measurement[0].id) 
+        const user = await User.query(trx).findById(userId)
+        const relatedIngredient = await user.$relatedQuery("ingredients", trx).relate({
+          id: existingIngredient.id,
+          amount: cleanedIngredient.amount,
+          unit: cleanedIngredient.unit,
+          description: cleanedIngredient.description
+        })
+        return relatedIngredient
+      })
+      const updatedIngredients = await User.relatedQuery("ingredients").for(userId).where({ name: cleanedIngredient.name })
+      const serializedIngredient = IngredientSerializer.getIngredientWithDetails(updatedIngredients[0])
+      return res.status(201).json({ editedIngredient: serializedIngredient })
+    }
+
+    const fetchedIngredient = await User.transaction(async (trx) => {
+      const measurement = await User.relatedQuery("pantryMeasurements", trx).for(userId).where("ingredientId", ingredientId)
+      const deleted = await PantryMeasurement.query(trx).deleteById(measurement[0].id)
+      const user = await User.query(trx).findById(userId)
+      const fetchedIngredient = await user.$relatedQuery("ingredients", trx).insertAndFetch(cleanedIngredient)
+      return fetchedIngredient
+    })
+    return res.status(201).json({ editedIngredient: fetchedIngredient})
+  } catch(err) {
+    if (err instanceof ValidationError || err instanceof NotNullViolationError) {
+      return res.status(422).json({ errors: err.data})
+    }
+    return res.status(500).json({ err })
+  }
+})
+
+ingredientsRouter.delete("/", async (req, res) => {
+  const { userId, ingredientId } = req.body
+  try {
+    const existingMeasurement = await User.relatedQuery("pantryMeasurements").for(userId).where("ingredientId", ingredientId)
+    const deleted = await PantryMeasurement.query().deleteById(existingMeasurement[0].id)
+    return res.status(203).json("successful deletion")
```

Comment:
I think proper status here would be 200 (request has succeeded) or 204 (request has succeeded, but that the client doesn't need to navigate away from its current page)